### PR TITLE
My Budget App

### PR DESCRIPTION
**Pull request by gsmalik030**
In This PR i created :

- Splash screen
- A simple page with the name of  app, and links to the sign up and log in pages.
- Sign up and log in pages
- The user can register in the app with full name, email and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Categories page
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the bottom of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- "Add a new category" page
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- "Add a new transaction" page
- The user fills out a form to create a new transaction with:
1. name (mandatory)
2. amount (mandatory)
3. categories 
- The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
- Created unit and integration tests for all the most important components of your RoR application.

[video documentation](https://www.loom.com/share/89d46890510c4f6c952948cde1037cfa)

### CODE REVIEW

File: app/views/entities/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <%= render entity %>
+  <% end %>
+</div>
+
+<div class= "total">
+<p>Total :</p>
+<p><%= @group.entities.sum(&:amount) %>$</p>
+</div>
+
+<%= link_to "New Transcation", new_group_entity_path(@group), class: "add-category btn btn__sign-in" %>
```
Comment (VICTOR-CHUKWU) [inline app/views/entities/index.html.erb:16]: - [ ] also as i mentioned in transaction kindly make this button to be at the bottom of the page as it is in the design :+1:

File: app/views/groups/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+<h1>Categories</h1>
+
+<div id="groups" class="categories">
+  <% @groups.each do |group| %>
+    <%= render group %>
+  <% end %>
+  <%= link_to "Add New Category", new_group_path, class: "add-category btn btn__sign-in" %>
+</div>
+
+</section>
```
Comment (VICTOR-CHUKWU) [inline app/views/groups/index.html.erb:12]: - [ ] The button is at the bottom of the page in the design, kindly fix this
in your design :point_down: 
Screenshot from 2023-06-01 11-34-23
given design :point_down: 
Screenshot from 2023-04-28 11-55-45

File: spec/models/entity_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  it 'should have valid name' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'should have valid amount' do
+    subject.amount = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/models/entity_spec.rb:20]: - [ ] I am adding this here but I am referring to all your tests, it is a good practice to add an integration test for your controller, I did not see any test for your controller or your view, kindly add some tests for them

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @gsmalik030 , 
## STATUS: CHANGES REQUIRED :recycle: 
Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by addod19** [comment]
### Hi @gsmalik030, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because of the following reasons:
- [ ] No linters setup for the project
- [ ] You did not follow the correct Git-flow workflow, thus creating the PR from a descriptive feature branch like for example; from `budget-app` branch into `dev` branch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**